File: SuperiorPackGroup/BLL/PalletControlReportBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class PalletControlReportBLL

    Private m_PalletControlReportTableAdapter As PalletControlReportTableAdapter = Nothing

    Public ReadOnly Property Adapter() As PalletControlReportTableAdapter
        Get
            If (Me.m_PalletControlReportTableAdapter Is Nothing) Then
                Me.m_PalletControlReportTableAdapter = New PalletControlReportTableAdapter
                Me.m_PalletControlReportTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return Me.m_PalletControlReportTableAdapter
        End Get
    End Property

    Public Function GetPalletControlReport() As SPGReports.PalletControlReportDataTable

        Return Adapter.GetPalletControlReport()
        
    End Function

End Class

File: SuperiorPackGroup/BLL/PalletControlReportBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class PalletControlReportBLL
	{

		private PalletControlReportTableAdapter m_PalletControlReportTableAdapter = null;

		public PalletControlReportTableAdapter Adapter
		{
			get
			{
				if (m_PalletControlReportTableAdapter == null)
				{
					m_PalletControlReportTableAdapter = new PalletControlReportTableAdapter();
					m_PalletControlReportTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return m_PalletControlReportTableAdapter;
			}
		}

		public SPGReports.PalletControlReportDataTable GetPalletControlReport()
		{

			return Adapter.GetPalletControlReport();

		}

	}

}